craft_providers/base.py

```diff
@@ -1029,6 +1082,8 @@ def warmup(
         self._image_check(executor=executor)
         self._post_image_check(executor=executor)
 
+        self._mount_shared_cache_dirs(executor=executor)
```

Comment:
@mr-cal do we need to do this here if done in setup?

Reply:
We need to do it every time since I believe the mount is removed when the instance is stopped.  

Sheng, do you know if the mounts are removed when lxd and multipass instances are stopped?

Reply:
The way we create the mount does not persistent mount/disk in the config. So it has to be remount every time it starts.